frontend/src/metabase-lib/v1/expressions/recursive-parser.js

```diff
@@ -58,17 +58,17 @@ function recursiveParse(source) {
 
   // Parameters ::= "(" * Expression ")"
   const parseParameters = () => {
-    expectOp(OP.OpenParenthesis);
+    expectOp(OP.OpenParenthesis, t`opening parenthesis`);
     const params = [];
     while (!matchOps([OP.Comma, OP.CloseParenthesis])) {
       const expr = parseExpression();
       params.push(expr);
-      if (!matchOps([OP.Comma])) {
+      if (!matchOps([OP.Comma]) && matchOps([OP.CloseParenthesis])) {
```

Comment:
Without this condition parser would throw "Expecting closing parenthesis" error which is why the issue was reported.